sw/host/hsmtool/acorn/acorn.rs

```diff
@@ -271,7 +271,7 @@ impl Acorn {
         unsafe {
             let public_key = acorn_bindgen::acorn_buffer {
                 // Transmute because the acorn API wants a mut ptr (but it wont mutate).
-                ptr: std::mem::transmute(public_key.as_ptr()),
+                ptr: std::mem::transmute::<*const u8, *mut u8>(public_key.as_ptr()),
```

Comment:
Why is this change needed? More over, this should be `pubic_key.as_ptr().cast_mut()`?

Reply:
Clippy asked for it, but yes you’re right I should use that method